src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbBase.java

```diff
@@ -332,22 +340,40 @@ public ListTablesResult listTables(ListTablesRequest listTablesRequest) {
 
     @Override
     public ListTablesResult listTables() {
-        return getAmazonDynamoDb().listTables();
+        return listTables((String)null);
     }
 
     @Override
     public ListTablesResult listTables(String exclusiveStartTableName) {
-        throw new UnsupportedOperationException();
+        return listTables(exclusiveStartTableName, 100);
     }
 
     @Override
     public ListTablesResult listTables(String exclusiveStartTableName, Integer limit) {
-        throw new UnsupportedOperationException();
+        if (mtContext.getContextOpt().isEmpty()) {
+            List<String> tableNames = Lists.newArrayList();
+            String innerExclusiveStartTableName = exclusiveStartTableName;
+
+            // filter out any physical tables on this account that this mt-dynamo instance does not manage
+            // and eagerly pull more pages until we fill up a full result set or run out of tables
+            do  {
+                ListTablesResult rawResults = getAmazonDynamoDb().listTables(innerExclusiveStartTableName, limit);
+                tableNames.addAll(
+                    rawResults.getTableNames().stream().filter(this::isMtTable).collect(Collectors.toList()));
+                innerExclusiveStartTableName = rawResults.getLastEvaluatedTableName();
+            } while (!(tableNames.size() >= limit || innerExclusiveStartTableName == null));
+
+            return new ListTablesResult()
+                .withLastEvaluatedTableName(tableNames.isEmpty() ? null : tableNames.get(tableNames.size() - 1))
```

Comment:
Shouldn't this always be null? (Since you are going through all tables at once.)

Reply:
it'll be non-null if tableNames fills its result set (limit tables scanned), which this test case exercises: https://github.com/salesforce/mt-dynamo/pull/411/files/c1163ffff0dfaea05d8c86e6a0bfed921e115c69#diff-ecb914953bc1a215b4e5c00fdeb28cafR30

Reply:
Then shouldn’t the condition be for whether to set last evaluated table name the same as in the next line? (Ie set to non-null if the names exceed limit)

Reply:
good catch, this doesn't match API contract of leaving lastEvaluatedTableName to null if there's no more tables to return. Will include that in a follow-up PR as well

Reply:
addressed this in https://github.com/salesforce/mt-dynamo/pull/472